subprojects/analysis-kotlin-api/src/test/kotlin/org/jetbrains/dokka/analysis/test/jvm/java/JavaAnalysisTest.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package org.jetbrains.dokka.analysis.test.jvm.java
+
+import org.jetbrains.dokka.analysis.test.api.javaTestProject
+import org.jetbrains.dokka.analysis.test.api.parse
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class JavaAnalysisTest {
+
+    @Test
+    fun javaTestProject() {
+        val testProject = javaTestProject {
+            dokkaConfiguration {
+                moduleName = "java-module-name-for-unit-test"
+            }
+            javaFile(pathFromSrc = "org/jetbrains/dokka/test/java/Bar.java") {
+                +"""
```

Comment:
I understand it might lead to some issues down the road, but I couldn't think of any substantial problems to justify changing it as our use case is rather simple. 

This API is not public, so we'll have a chance to change it quickly if it's ever published externally.